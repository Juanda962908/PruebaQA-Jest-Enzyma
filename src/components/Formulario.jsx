import React from 'react'
import { FormControl,InputLabel,Input,Typography,FormHelperText,Checkbox,FormControlLabel, Box,Button} from '@material-ui/core';
import { makeStyles } from '@material-ui/core/styles';


const useStyle = makeStyles({

formulario:{
display:'flex',
flexDirection:'column',
justifyContent:'space-evenly',
alignItems:'space-evenly'
},
camposFormulario :{
color:'white',
width:'20rem'
},
botonEnviar:{
  backgroundColor:'rgb(24, 34, 60,90%)',
  marginTop:'2vh',
  width:'10vw',
  '&:hover': {
  backgroundColor: 'rgb(24, 34, 60,100%)',
  borderColor: '#0062cc',
  boxShadow: 'none',
  },
  },
textoLabel:{
  fontSize:'1rem',
  
}

})





function Formulario(props) {
const classes = useStyle();
return (
<>
<Typography variant='h3'>PRUEBA TECNICA QA - JUAN DANIEL DE LA HOZ F.</Typography>
  <form action="" marginTop="10rem">
<Box display='flex' flexDirection='column' marginTop="15rem">
    <FormControl>
      <InputLabel className={classes.camposFormulario} htmlFor="my-input">Nombre Completo</InputLabel>
      <Input className={classes.camposFormulario} id="Input1" aria-describedby="my-helper-text"
        onChange={props.handleChange}  />
      <FormHelperText className={classes.camposFormulario} id="my-helper-text">
      </FormHelperText>
    </FormControl>
    <FormControl>
      <InputLabel className={classes.camposFormulario} htmlFor="my-input">Apellido</InputLabel>
      <Input className={classes.camposFormulario} id={props.idInput2} aria-describedby="my-helper-text"
        onChange={props.handleChange} />
      <FormHelperText className={classes.camposFormulario} id="my-helper-text">
      </FormHelperText>
    </FormControl>

    <Box marginTop='2rem'>
      <Checkbox checked={props.checked} onChange={props.handleChange} inputProps={{ 'aria-label': 'primary checkbox' }} />
      <label className={classes.textoLabel} htmlFor="">{props.textoDeLabel}</label>
    </Box>

    <Box width='20rem' display='flex' flexDirection='row' alignItems='center' justifyContent='center'>
      <Button onClick={props.funcionButton} href="https://www.google.com/"  variant="contained" color="primary" className={classes.botonEnviar} type="submit">
        Enviar
      </Button>
      </Box>
    </Box>
  </form>
</>
)
}

export default Formulario