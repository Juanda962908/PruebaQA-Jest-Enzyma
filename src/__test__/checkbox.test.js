import React from 'react'
import renderer from 'react-test-renderer'
import Formulario from '../components/Formulario'
import {mount, shallow} from 'enzyme'
import enzymeConfig from '../enzymeConfig'
import {Checkbox,Input} from '@material-ui/core'

describe('<Checkbox/>', ()=>{
 
   
    it('Renderizar Formulario para que se genere Snapshot', ()=>{
        const component = renderer.create(<Formulario />);
        const tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });
    

    it('validar que contenga dos webElement checkbox',()=>{
        const component = shallow(<Formulario />);
        expect(component.find(Checkbox).length).toBe(1);
    })
    
     it('validar que contenga solo un webElement label',() => {
        const component = shallow(<Formulario />);
        expect(component.find('label').length).toBe(1);
    });   

     it('validar que el props que reciba el label sea igual a texto ',()=>{
     var texto = "Quiero recibir informaci√≥n";
     const component = shallow(<Formulario textoDeLabel={texto}/>);
     expect(component.find('label').text()).toEqual(texto);     
    });

   it('funcionamiento del onchange en el primer input', ()=>{
       const mockHandler = jest.fn();
       const component = shallow(<Formulario handleChange={mockHandler}/>);
       const input = component.find(Checkbox).at(0);
       input.checked = true;
       input.simulate('change');
       expect(mockHandler.mock.calls.length).toBe(1);   
   });
 
})