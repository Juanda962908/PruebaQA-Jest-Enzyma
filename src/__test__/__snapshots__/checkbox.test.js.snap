// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Checkbox/> Renderizar Formulario para que se genere Snapshot 1`] = `
Array [
  <h3
    className="MuiTypography-root MuiTypography-h3"
  >
    PRUEBA TECNICA QA - JUAN DANIEL DE LA HOZ F.
  </h3>,
  <form
    action=""
    marginTop="10rem"
  >
    <div
      className="MuiBox-root MuiBox-root-5"
    >
      <div
        className="MuiFormControl-root"
      >
        <label
          className="MuiFormLabel-root MuiInputLabel-root makeStyles-camposFormulario-2 MuiInputLabel-formControl MuiInputLabel-animated"
          data-shrink={false}
          htmlFor="my-input"
        >
          Nombre Completo
        </label>
        <div
          className="MuiInputBase-root MuiInput-root MuiInput-underline makeStyles-camposFormulario-2 MuiInputBase-formControl MuiInput-formControl"
          onClick={[Function]}
        >
          <input
            aria-describedby="my-helper-text"
            aria-invalid={false}
            className="MuiInputBase-input MuiInput-input"
            disabled={false}
            id="Input1"
            onAnimationStart={[Function]}
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            required={false}
            type="text"
          />
        </div>
        <p
          className="MuiFormHelperText-root makeStyles-camposFormulario-2"
          id="my-helper-text"
        />
      </div>
      <div
        className="MuiFormControl-root"
      >
        <label
          className="MuiFormLabel-root MuiInputLabel-root makeStyles-camposFormulario-2 MuiInputLabel-formControl MuiInputLabel-animated"
          data-shrink={false}
          htmlFor="my-input"
        >
          Apellido
        </label>
        <div
          className="MuiInputBase-root MuiInput-root MuiInput-underline makeStyles-camposFormulario-2 MuiInputBase-formControl MuiInput-formControl"
          onClick={[Function]}
        >
          <input
            aria-describedby="my-helper-text"
            aria-invalid={false}
            className="MuiInputBase-input MuiInput-input"
            disabled={false}
            onAnimationStart={[Function]}
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            required={false}
            type="text"
          />
        </div>
        <p
          className="MuiFormHelperText-root makeStyles-camposFormulario-2"
          id="my-helper-text"
        />
      </div>
      <div
        className="MuiBox-root MuiBox-root-6"
      >
        <span
          aria-disabled={false}
          className="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-7 MuiCheckbox-root MuiCheckbox-colorSecondary MuiIconButton-colorSecondary"
          onBlur={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={null}
        >
          <span
            className="MuiIconButton-label"
          >
            <input
              aria-label="primary checkbox"
              className="PrivateSwitchBase-input-10"
              data-indeterminate={false}
              onChange={[Function]}
              type="checkbox"
            />
            <svg
              aria-hidden={true}
              className="MuiSvgIcon-root"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
              />
            </svg>
          </span>
        </span>
        <label
          className="makeStyles-textoLabel-4"
          htmlFor=""
        />
      </div>
      <div
        className="MuiBox-root MuiBox-root-11"
      >
        <a
          aria-disabled={false}
          className="MuiButtonBase-root MuiButton-root MuiButton-contained makeStyles-botonEnviar-3 MuiButton-containedPrimary"
          href="https://www.google.com/"
          onBlur={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
        >
          <span
            className="MuiButton-label"
          >
            Enviar
          </span>
        </a>
      </div>
    </div>
  </form>,
]
`;
